src/waveform/renderers/waveformrendermark.cpp

```diff
@@ -160,6 +169,13 @@ void WaveformRenderMark::generateMarkImage(WaveformMark* pMark) {
             }
         }
 
+        // Visualize cue source (if available)
+        if (pMark->hasSource()) {
+            if (pMark->getSource() != Cue::MANUAL) {
```

Comment:
What happens if new cue sources appear in the future? Did you actually want to use the following condition:
 if (pMark->getSource() == Cue::UNKKNOWN) {
That would make sense to me if "?" is used as the label.